packages/middleware-token/src/tokenMiddleware.ts

```diff
@@ -0,0 +1,32 @@
+import { HttpRequest } from "@aws-sdk/protocol-http";
+import {
+  FinalizeHandler,
+  FinalizeHandlerArguments,
+  FinalizeHandlerOutput,
+  FinalizeRequestMiddleware,
+  RelativeMiddlewareOptions,
+} from "@aws-sdk/types";
+
+import { TokenResolvedConfig } from "./configurations";
+
+export const TokenMiddlewareOptions: RelativeMiddlewareOptions = {
+  name: "tokenMiddleware",
+  tags: ["TOKEN"],
+  relation: "after",
+  toMiddleware: "retryMiddleware",
+  override: true,
+};
+
+export const tokenMiddleware =
+  <Input extends object, Output extends object>(
+    options: TokenResolvedConfig
+  ): FinalizeRequestMiddleware<Input, Output> =>
+  (next: FinalizeHandler<Input, Output>): FinalizeHandler<Input, Output> =>
+    async function (args: FinalizeHandlerArguments<Input>): Promise<FinalizeHandlerOutput<Output>> {
```

Comment:
super nit: arrow function?
```suggestion
    (args: FinalizeHandlerArguments<Input>): Promise<FinalizeHandlerOutput<Output>> => {
```

Reply:
Done in [`b428374` (#3883)](https://github.com/aws/aws-sdk-js-v3/pull/3883/commits/b428374528a06ef7e4bc42eed9c80fe3b92a133e)